### DISCUSSION TITLE
AOC 2022 Day Eight

### DISCUSSION THREAD

**Discussion by danieltott** [Virtual-Coffee/virtualcoffee.io#766] [Advent of Code 2022]
**Discussion of solutions for [Advent of Code Day 8](https://adventofcode.com/2022/day/8) here.**

Don't forget about the VC Advent of Code private leaderboard:
- Go to [leaderboard/private](https://adventofcode.com/2022/leaderboard/private)
- Use join code `363496-e6f42283`

**Comment by danieltott** [comment]
NO FANCY! HULK HATE FANCY! ONLY `FOR` LOOPS!

https://github.com/danieltott/aoc2022/blob/main/src/day08/index.ts

**Reply by rhdeck** [reply]
You wouldn't like me when I'm fancy [(Stable Diffusion take on this)](https://playgroundai.com/post/clbf3pdu82v9ds601puf5r4fk)

**Reply by doleraj** [reply]
It is fascinating that you beat the absolute pants off my solution for runtime, even with the less-fancy option for solving it. I suppose that makes sense given that I've got O(n^diggity-boo) going on with mine.

**Comment by rhdeck** [comment]
Zip and list comprehension again to my rescue. https://github.com/rhdeck/aoc-2022/blob/main/src/8/day_8.ipynb

**Reply by tkshill** [reply]
Perfectly functional, as all comprehensions should be

**Comment by doleraj** [comment]
Slightly more fancy than for loops, but only just barely: https://github.com/doleraj/adventofcode/blob/main/aoc2022/src/day08/index.ts

plz no smash

**Reply by tkshill** [reply]
No hulk smash here. Just slicing and dicing!

**Comment by tkshill** [comment]
Functions and some scripts and silly mistakes. I'm satisfied. Mainly because I think I made full use of the standard libraries.

[solver here](https://github.com/tkshill/AdventOfCode2022/blob/main/AOC2022/Day08.fs)

edit: wrote my only converter from seq to 2d array just to discover one already existed. So there's still a lot I need to learn from the standard library.
